src/compiler/transformers/ts.ts

```diff
@@ -2034,6 +2059,11 @@ namespace ts {
             );
         }
 
+        function isSimpleInlineableExpression(expression: Expression) {
+            return !isIdentifier(expression) && isSimpleCopiableExpression(expression) ||

```

Comment:
Would a `GeneratedIdentifier` be "inlineable"?

Reply:
Only if such a `Generatedidentifier` has all of its assignments occur before any intended usages of the identifier. Not something we check for in a uniform way, I think, but it may happen to be true for all generated identifiers at present?